src/Admin/VMware.ScriptRuntimeService.AdminEngine/VCRegistration/VCTrustedCertificatesCollector.cs

```diff
@@ -51,21 +59,71 @@ public LoggingHandler(ILoggerFactory loggerFactory, HttpMessageHandler innerHand
          }
 
          protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) {
-            _logger.LogDebug($"Request: {request}");
-
-            if (request.Content != null) {
-               _logger.LogDebug(await request.Content.ReadAsStringAsync());
-            }
+            _logger.LogDebug($"Request: {await ToLogMessage(request)}");
 
             HttpResponseMessage response = await base.SendAsync(request, cancellationToken);
 
-            _logger.LogDebug($"Response: {response}");
-            if (response.Content != null) {
-               _logger.LogDebug(await response.Content.ReadAsStringAsync());
-            }
+            _logger.LogDebug($"Response: {await ToLogMessage(response)}");
 
             return response;
          }
+
+         private async static Task<string> ToLogMessage(HttpResponseMessage response) {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.Append("StatusCode: ");
+            stringBuilder.Append((int) response.StatusCode);
+            stringBuilder.Append(", ReasonPhrase: '");
+            stringBuilder.Append(response.ReasonPhrase ?? "<null>");
+            stringBuilder.Append("', Version: ");
+            stringBuilder.Append(response.Version);
+            stringBuilder.Append(", Content: ");
+            stringBuilder.Append((response.Content == null) ? "<null>" : await response.Content.ReadAsStringAsync());
+            stringBuilder.Append(", Headers:\r\n");
+            stringBuilder.Append(DumpHeaders(Enumerable.Empty<string>(), response.Headers, (response.Content == null) ? null : response.Content.Headers));
+            return stringBuilder.ToString();
+         }
+
+         private async static Task<string> ToLogMessage(HttpRequestMessage request) {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.Append("Method: ");
+            stringBuilder.Append(request.Method);
+            stringBuilder.Append(", RequestUri: '");
+            stringBuilder.Append((request.RequestUri == null) ? "<null>" : request.RequestUri.ToString());
+            stringBuilder.Append("', Version: ");
+            stringBuilder.Append(request.Version);
+            stringBuilder.Append(", Content: ");
+            stringBuilder.Append((request.Content == null) ? "<null>" : await request.Content.ReadAsStringAsync());
+            stringBuilder.Append(", Headers:\r\n");
+            stringBuilder.Append(DumpHeaders(new[] { "Authorization" }, request.Headers, (request.Content == null) ? null : request.Content.Headers));
+            return stringBuilder.ToString();
+         }
+
+         private static string DumpHeaders(IEnumerable<string> headersToSanitize, params HttpHeaders[] headers) {
```

Comment:
Add a unit test for this method so we don't unintentionally break this over time especially since I doubt that in the future we will be allowed to do integration tests for the logging functionality.

Reply:
I added few tests